### PR TITLE
replace eth-sig-utils and ethereumjs-utils with ethers in locksmith

### LINKED ISSUE #9057: Replace eth-sig-util with ethers in locksmith
We use [eth-sig-util](https://togithub.com/MetaMask/eth-sig-util) in locksmith because at the time, ethers did not have a way to sign typed data.... but `ethers` now supports it. We should remove [eth-sig-util](https://togithub.com/MetaMask/eth-sig-util) and use ethers.

### PR DESCRIPTION
**Pull request by m00n620**
# Description

Replace eth-sig-utils and ethereum-js with ethers in locksmith workspace.
This PR should close #9057 and #9058

# Issues

Fixes #9057 and #9058

# Checklist:
- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/__tests__/signatureValidation/signPersonal.test.ts [hunk 1]
```diff
@@ -75,7 +69,7 @@ beforeAll(() => {
 describe('Signature Validation Middleware', () => {
   describe('generateSignatureEvaluator', () => {
     describe('when the request has a token', () => {
-      it('returns the signee', (done) => {
+      it('returns the signee', async () => {
```
Comment (julien51) [inline locksmith/__tests__/signatureValidation/signPersonal.test.ts:72]: good catch :) This is cleaner with Promises!

File: locksmith/__tests__/signatureValidation/signPersonal.test.ts [hunk 2]
```diff
@@ -131,7 +126,7 @@ describe('Signature Validation Middleware', () => {
 
   describe('when a valid signature is received', () => {
     describe('a signature for User creation', () => {
-      it('moves the request to the application', (done) => {
+      it('moves the request to the application', async () => {
```
Comment (julien51) [inline locksmith/__tests__/signatureValidation/signPersonal.test.ts:129]: same

**Review by julien51 — COMMENTED** [review]
First pass done! This is looking very promising! I'll do another check with fresher eyes tomorrow :)

**Comment by m00n620** [comment]
> 

Fixing some bugs right now

**Comment by m00n620** [comment]
@julien51 , finally done, passed all test cases

File: locksmith/__tests__/controllers/lockController.test.js [hunk 3]
```diff
@@ -2,44 +2,35 @@ const request = require('supertest')
...
 const validLockOwner = '0xAaAdEED4c0B861cB36f4cE006a9C90BA2E43fdc2'
 const validLockAddress = '0x21cC9C438D9751A3225496F6FD1F1215C7bd5D83'
 const chain = 31337
 
 const testLockDetails = {
   chain,
   name: 'Test Lock',
-  address: '0xab7c74abC0C4d48d1bdad5DCB26153FC8780f83E',
-  owner: '0xCA750f9232C1c38e34D27e77534e1631526eC99e',
+  address: '0x21cC9C438D9751A3225496F6FD1F1215C7bd5D83',
```
Comment (julien51) [inline locksmith/__tests__/controllers/lockController.test.js:13]: Any reason you changed these?

File: locksmith/__tests__/controllers/lockController.test.js [hunk 4]
```diff
@@ -2,44 +2,35 @@ const request = require('supertest')
...
+  owner: '0xAaAdEED4c0B861cB36f4cE006a9C90BA2E43fdc2',
```
Comment (julien51) [inline locksmith/__tests__/controllers/lockController.test.js:14]:
```suggestion
  owner: validLockOwner,
```

File: locksmith/__tests__/controllers/lockController.test.js (hunk 3 above)
Comment (julien51) [inline locksmith/__tests__/controllers/lockController.test.js:13]:
```suggestion
  address: validLockAddress,
```

File: locksmith/__tests__/controllers/metadataController/updateDefaults.test.ts [hunk 5]
```diff
@@ -60,22 +53,26 @@ function generateTypedData(message: any) {
     },
     primaryType: 'LockMetadata',
     message,
+    messageKey,
```
Comment (julien51) [inline locksmith/__tests__/controllers/metadataController/updateDefaults.test.ts:56]: Do we need this?

File: locksmith/__tests__/test-helpers/typeDataGenerators.ts [hunk 6]
```diff
@@ -42,5 +28,6 @@ export function keyTypedData(message: any) {
     },
     primaryType: 'KeyMetadata',
     message,
+    messageKey,
```
Comment (julien51) [inline locksmith/__tests__/test-helpers/typeDataGenerators.ts:31]: Do we need this?

File: locksmith/scripts/metadataUpload/metadata_upload.ts [hunk 7]
```diff
@@ -112,11 +89,11 @@ async function main(
 //'/Users/akeem/projects/unlock/locksmith/scripts/data.json'
 //'http://localhost:8080'
 
-let privateKey = args.privateKey
-let lockAddress = args.lockAddress
-let host = args.host
-let inputFile = args.inputFile
-let scope = args.scope
+const privateKey = args.privateKey
```
Comment (julien51) [inline locksmith/scripts/metadataUpload/metadata_upload.ts:92]: Thanks for fixing these!

File: locksmith/scripts/metadataUpload/user_metadata_upload.ts [hunk 8]
```diff
@@ -51,6 +35,7 @@ function generateUserMetadataPayload(message: any) {
     },
     primaryType: 'UserMetaData',
     message: message,
+    messageKey,
```
Comment (julien51) [inline locksmith/scripts/metadataUpload/user_metadata_upload.ts:38]: Same here, I am not sure we need these extra params?

File: locksmith/src/controllers/lockController.js [hunk 9]
```diff
@@ -29,8 +29,8 @@ const lockSave = async (req, res) => {
   if (!databaseLock) {
     lock.chain = req.chain
     await createLock(lock)
-    return res.sendStatus(200)
   }
+  return res.sendStatus(200)
```
Comment (julien51) [inline locksmith/src/controllers/lockController.js:33]: Oh nice catch!

**Review by julien51 — COMMENTED** [review]
Good one! A gew comments but I think we can complete these very quickly! Thank you!

File: locksmith/__tests__/controllers/lockController.test.js (hunk 4 above)
Comment (m00n620) [reply locksmith/__tests__/controllers/lockController.test.js]: resolved

File: locksmith/__tests__/controllers/lockController.test.js (hunk 3 above)
Comment (m00n620) [reply locksmith/__tests__/controllers/lockController.test.js]: resolved

Comment (m00n620) [reply locksmith/__tests__/controllers/lockController.test.js]: resolved

**Review by julien51 — APPROVED** [review]
LGTM!

**Comment by julien51** [comment]
Great, thanks a LOT! I will merge it and pay the bounty!

**Pull request merged**